### PR TITLE
Layout - Object scanning should only check class-attributes for LayoutRenderers

### PR DESCRIPTION
**Pull request by snakefoot**
Only Layouts and LayoutRenderers are relevant. Should ignore NLogConfigurationItem-objects.

It is confusing for NLog users that they need to add additional special attributes to custom-classes marked with NLogConfigurationItem (When creating their own custom targets/layouts). If forgetting then NLog-Layout-engine will see everything as "unsafe".

### CODE REVIEW

File: src/NLog/Layouts/JsonAttribute.cs [hunk 1]
```diff
@@ -40,9 +40,6 @@ namespace NLog.Layouts
     /// JSON attribute.
     /// </summary>
     [NLogConfigurationItem]
-    [ThreadAgnostic]
-    [ThreadSafe]
-    [AppDomainFixedOutput]
```
Comment (304NotModified) [inline src/NLog/Layouts/JsonAttribute.cs:45]: So this is not needed anymore for all attributes?

Comment (snakefoot) [reply src/NLog/Layouts/JsonAttribute.cs:45]: No objects that are just marked with `NLogConfigurationItem` (To signal they are safe for NLog to probe) doesn't need the extra attributes. The extra attributes are only for Layout and LayoutRenderers.

Comment (304NotModified) [reply src/NLog/Layouts/JsonAttribute.cs:45]: Could we please add that to the docs of those attributes? Then I will merge it 

(please no force push)

Comment (snakefoot) [reply src/NLog/Layouts/JsonAttribute.cs:45]: Not sure I understand. Should I write that attributes intended for Layout and LayoutRenderers should only need to be used for Layout and LayoutRenderers ?

Comment (304NotModified) [reply src/NLog/Layouts/JsonAttribute.cs:45]: Ah it was already there. Surprising!

**Pull request merged**